libraries/splunk_logs.rb

```diff
@@ -0,0 +1,122 @@
+# coding: UTF-8
+
+# Cookbook Name:: cerner_splunk
+# File Name:: splunk_logs.rb
+#
+# Libraries for configuring splunk's internal log settings.
+
+require 'chef/provider'
+require 'chef/resource'
+
+class Chef
+  class Resource
+    # Chef Resource for managing Splunk log config.
+    class SplunkLogs < Chef::Resource
+      provides :splunk_logs if respond_to?(:provides)
+
+      def initialize(name, run_context = nil)
+        super
+        @resource_name = :splunk_logs
+        @action = :create
+        @allowed_actions = %i[create remove]
+        @contents = {}
+      end
+
+      def contents(arg = nil)
+        set_or_return(:contents, arg, kind_of: Hash)
+      end
+    end
+  end
+end
+
+require_relative 'conf'
+
+class Chef
+  class Provider
+    # Chef Provider for managing Splunk log config
+    class SplunkLogs < Chef::Provider # rubocop:disable ClassLength
+      provides :splunk_logs if respond_to?(:provides)
+
+      def whyrun_supported?
+        true
+      end
+
+      def load_current_resource
+        @current_resource ||= Chef::Resource::SplunkLogs.new(new_resource.name)
+      end
+
+      def action_create
+        manage_file("/opt/splunk/etc/log-local.cfg", {"splunkd": new_resource.contents})
+      end
+
+      # uninstall the app by removing the apps directory
+      def action_remove
+        file = Chef::Resource::Template.new("/opt/splunk/etc/log-local.cfg", run_context)
+        file.path("/opt/splunk/etc/log-local.cfg")
+        file.run_action(:delete)
+        new_resource.updated_by_last_action(file.updated_by_last_action?)
+      end
+
+      def delete_file(file_path)
+        download = Chef::Resource::File.new(file_path, run_context)
+        download.run_action(:delete)
+      end
+
+      def symbolize_keys(hash)
```

Comment:
The manage_file/insert_procs/hash_to_proc/symbolize_keys functions are copied from libraries/splunk_apps.rb. Would it make sense to create a shared utility to host these functions, and if so what would be the right place to put it?

Reply:
Do we really need this separate resource file or can we reuse the splunk_template resource in the recipe instead? e.g. 
```
splunk_template 'etc/log-local.cfg' do
  stanzas log_local_contents
  notifies :touch, 'file[splunk-marker]', :immediately
end
```

Reply:
As I was experimenting to learn the structure of the repo, the approaches I was trying didn't fit the mold of the template, and it didn't occur to me that they might now.

So would your suggestion be to add log-local.cfg to the list of KNOWN_CONFIG_FILES and add a regex for it to the case-when block in the template's initialize method?

Reply:
Yes, that would be my suggestion.

Reply:
I've done this here: https://github.com/cerner/cerner_splunk/pull/191/commits/48df09d9b795bd2b1ec22fc86b6345a788aaed42